stl/inc/ranges

```diff
@@ -880,11 +893,10 @@ namespace ranges {
             }
         }
 
-        // clang-format off
-        _NODISCARD friend constexpr bool operator==(
-            const _Ioterator&, const _Ioterator&) requires equality_comparable<_Wi> = default;
-        // clang-format on
-
+        _NODISCARD friend constexpr bool operator==(const _Ioterator& _Left, const _Ioterator& _Right) noexcept(
+            noexcept(_Left._Current == _Right._Current)) /* strengthened */ requires equality_comparable<_Wi> {
+            return _Left._Current == _Right._Current;
+        }

```

Comment:
I think there is an issue with the base classes. This was breaking due to an deleted function.

Now that I think about it it might work when we default `operator<=>` for the bases?

Reply:
Ah, that's it: we added bases, they need to support `==` for this class to default it. This is probably also breaking my ABI-aggregate property =( Let me dig into this.

Reply:
What's the lesser evil here:
1. Lose the ability to pass `iota_view::iterator` in registers
1. Have two partial specializations of `iota_view::iterator`, duplicates except that only one has `iterator_category`.
1. Use the preprocessor to stamp out two specializations of `iota_view::iterator`, identical except for the presence of `iterator_category`

Reply:
FYI: after a bit of discussion offline with miscco and STL, I've given up un ABI-aggregateness. I'll push a change to add back the constructors I had miscco remove, default `==` and `<=>` everywhere possible, and cleanup all of the initializers that were changed from `I{x}` to `I{{}, x}` when we were trying to preserve the aggregate property.